source/OptimaJet.DWKit.StarterApplication/Repositories/UserRepository.cs

```diff
@@ -6,17 +6,45 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
 using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Services;
 
 namespace OptimaJet.DWKit.StarterApplication.Repositories
 {
     public class UserRepository : DefaultEntityRepository<User>
     {
+        public IOrganizationContext OrganizationContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
         public UserRepository(
             ILoggerFactory loggerFactory,
             IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
             IDbContextResolver contextResolver
             ) : base(loggerFactory, jsonApiContext, contextResolver)
         {
+            this.OrganizationContext = organizationContext;
+            this.CurrentUserContext = currentUserContext;
+        }
+
+        public override IQueryable<User> Get()
+        {
+            if (OrganizationContext.InvalidOrganization) return base.Get().Take(0);
+            if (!OrganizationContext.HasOrganization)
+            {
+                // No organization specified, so include all users in the all the organizations that the current user is a member
+                var currentUser = GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+                var currentUserOrgIds = currentUser.OrganizationMemberships.Select(o => o.OrganizationId);
+                return base.Get()
+                           .Include(u => u.OrganizationMemberships)
+                           .Where(u => u.OrganizationMemberships.Select(o => o.OrganizationId).Intersect(currentUserOrgIds).Any());
+            }
+            else
+            {
```

Comment:
because we are returning in each block, we can omit the `else` part and reduce the number of indentations :)

Reply:
OK